react/features/filmstrip/components/native/Thumbnail.js

```diff
@@ -57,19 +58,27 @@ class Thumbnail extends Component {
      */
     render() {
         const audioTrack = this.props._audioTrack;
+        const disablePin = this.props.disablePin;
         const largeVideo = this.props._largeVideo;
         const participant = this.props.participant;
         const videoTrack = this.props._videoTrack;
 
         let style = styles.thumbnail;
 
-        if (participant.pinned) {
+        if (participant.pinned && !disablePin) {
             style = {
                 ...style,
                 ...styles.thumbnailPinned
             };
         }
 
+        if (this.props.styleOverrides) {
+            style = {
```

Comment:
why not use style arrays? That way it doesn't matter if styleOverrides is undefined.